.gitmodules

```diff
@@ -7,3 +7,6 @@
 [submodule "lib/permit2"]
 	path = lib/permit2
 	url = https://github.com/Uniswap/permit2.git
+[submodule "lib/create3-factory"]
+	path = lib/create3-factory
+	url = https://github.com/ZeframLou/create3-factory.git
```

Comment:
Am I correct in assuming that when using create3Factory, the expectation is to use the same deployer address for a contract across all chains? This implies that the deployer address should remain consistent.

For instance, if we use the address `0x123` to deploy `Router` contract on multiple chains, then decide to deploy it to a new chain after a year, we must use the same `0x123` address to maintain the `Router` contract's address consistency.

Are we supposed to be able to use `0xabc` to deploy the `Router` contract to the same address?

Reply:
IMO we have to use the same address which depolyed create3Factory on original chain to deploy create3Factory to new chain. Then we could use different address to deploy `Router` to new chain.  Am I correct ?

Reply:
We need another factory deployer address to deploy CREATE3Factory to ensure the address of CREATE3Factory is the same on multiple chains.  The factory deployer should deploy CREATE3Factoy only to make sure the nonce is 0.

For easier management, it is advisable to use a separate factory deployer different from the contract deployer responsible for deploying the router, callback, and fee contracts. 

To simplify the testing of the deployment script in a CI environment, I utilize the same deployer for all contract deployments.